### ISSUE TITLE
Pytorch module is not initialized when training on bigger dataset on 9070xt

### ISSUE THREAD

**Issue by Yxeyuo** [ROCmSoftwarePlatform/pytorch#2240]
### 🐛 Describe the bug

When training on a bigger dataset  on a 9070xt (bigger means around 15k images) python exits with the error:
:0:/longer_pathname_so_that_rpms_can_support_packaging_the_debug_info_for_all_os_profiles/src/clr/hipamd/src/hip_global.cpp:158 : 15742695212 us:  Module not initialized

I wrote a training script for the dfire dataset: https://github.com/gaiasd/DFireDataset?tab=readme-ov-file to show the error
`import os
import torch
import torch.optim as optim
from torch.amp import autocast, GradScaler
from torch.utils.data import Dataset, DataLoader
from PIL import Image
import torchvision
from torchvision import transforms
from torchvision.models.detection.faster_rcnn import FastRCNNPredictor

class YOLODataset(Dataset):
    def __init__(self, images_dir, labels_dir, transform=None):
        self.images_dir = images_dir
        self.labels_dir = labels_dir
        self.transform = transform
        self.image_files = [
            f for f in os.listdir(images_dir)
            if f.lower().endswith((".jpg", ".png", ".jpeg"))
        ]

    def __len__(self):
        return len(self.image_files)

    def __getitem__(self, idx):
        img_name = self.image_files[idx]
        img_path = os.path.join(self.images_dir, img_name)
        label_path = os.path.join(self.labels_dir, os.path.splitext(img_name)[0] + ".txt")

        image = Image.open(img_path).convert("RGB")
        w, h = image.size

        boxes = []
        labels = []
        if os.path.exists(label_path):
            with open(label_path) as f:
                for line in f:
                    parts = line.strip().split()
                    if len(parts) != 5:
                        continue
                    cls, x_center, y_center, bw, bh = map(float, parts)
                    x_c = x_center * w
                    y_c = y_center * h
                    bw_abs = bw * w
                    bh_abs = bh * h
                    xmin = x_c - bw_abs / 2
                    ymin = y_c - bh_abs / 2
                    xmax = x_c + bw_abs / 2
                    ymax = y_c + bh_abs / 2
                    boxes.append([xmin, ymin, xmax, ymax])
                    labels.append(int(cls) + 1)

        if len(boxes) == 0:
            boxes = torch.zeros((0, 4), dtype=torch.float32)
            labels = torch.zeros((0,), dtype=torch.int64)
        else:
            boxes = torch.tensor(boxes, dtype=torch.float32)
            labels = torch.tensor(labels, dtype=torch.int64)

        image_tensor = transforms.ToTensor()(image)

        target = {
            "boxes": boxes,
            "labels": labels,
            "image_id": torch.tensor([idx])
        }

        return image_tensor, target

def get_model(num_classes):
    model = torchvision.models.detection.fasterrcnn_resnet50_fpn(weights="COCO_V1")
    in_features = model.roi_heads.box_predictor.cls_score.in_features
    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, num_classes)
    return model

def collate_fn(batch):
    return tuple(zip(*batch))

def main():
    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    train_images = "./dfire/train/images"
    train_labels = "./dfire/train/labels"
    val_images = "./dfire/test/images"
    val_labels = "./dfire/test/labels"

    train_dataset = YOLODataset(train_images, train_labels, transform=None)
    val_dataset = YOLODataset(val_images, val_labels, transform=None)

    train_loader = DataLoader(
        train_dataset,
        batch_size=4,
        shuffle=True,
        num_workers=4,
        pin_memory=True,
        collate_fn=collate_fn
    )
    val_loader = DataLoader(
        val_dataset,
        batch_size=4,
        shuffle=False,
        num_workers=4,
        pin_memory=True,
        collate_fn=collate_fn
    )

    num_classes = 2
    model = get_model(num_classes).to(device)
    params = [p for p in model.parameters() if p.requires_grad]
    optimizer = optim.Adam(params, lr=1e-4)
    scaler = GradScaler()
    epochs = 10

    for epoch in range(epochs):
        model.train()
        for images, targets in train_loader:
            images = [img.to(device) for img in images]
            targets = [{k: v.to(device) for k, v in t.items()} for t in targets]

            optimizer.zero_grad()
            with autocast(device_type=device.type):
                loss_dict = model(images, targets)
                losses = sum(loss for loss in loss_dict.values())

            scaler.scale(losses).backward()
            scaler.step(optimizer)
            scaler.update()

        model.eval()
        with torch.no_grad():
            total_loss = 0.0
            for images, targets in val_loader:
                images = [img.to(device) for img in images]
                targets = [{k: v.to(device) for k, v in t.items()} for t in targets]
                with autocast(device_type=device.type):
                    loss_dict = model(images, targets)
                    losses = sum(loss for loss in loss_dict.values())
                total_loss += losses.item()
            avg_val_loss = total_loss / len(val_loader)
        print(f"Epoch {epoch+1}/{epochs}, Validation Loss: {avg_val_loss:.4f}")

    torch.save(model.state_dict(), "fasterrcnn_dfire.pth")

if __name__ == "__main__":
    main()
`
the error also happens while using ultralytics yolo. when i turn of amp in yolo training the first epoch will be successful but the validation step will crash the training. 
import torch, os
from ultralytics import YOLO

print("PyTorch-Version:", torch.__version__)
print("CUDA verfügbar:", torch.cuda.is_available())
if torch.cuda.is_available():
    print("  • GPU0:", torch.cuda.get_device_name(0))

model = YOLO("yolov8n.pt")
model.to("cuda")

data_yaml = "dfire/config.yaml"
epochs    = 20
imgsz     = 320      
batch     = 256       
run_name  = "dfire_experiment_fast"

results = model.train(
    data        = data_yaml,
    epochs      = epochs,
    imgsz       = imgsz,
    batch       = batch,
    name        = run_name,
    device      = "cuda",
    workers     = 6,       
    amp         = False,    
    verbose     = False,    
    plots       = False,    
    save_period = 1         
)

print("\nTraining abgeschlossen!")
ckpt_dir  = os.path.join("runs", "train", run_name)
last_ckpt = os.path.join(ckpt_dir, "weights", "last.pt")
best_ckpt = os.path.join(ckpt_dir, "weights", "best.pt")
print(f"→ Bestes Modell:   {best_ckpt}")
print(f"→ Finale Gewichte: {last_ckpt}")

### Versions

Collecting environment information...
PyTorch version: 2.6.0+git684f6f2
Is debug build: False
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: 6.4.43483-a187df25c

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 19.0.0git (https://github.com/RadeonOpenCompute/llvm-project roc-6.4.1 25184 c87081df219c42dc27c5b6d86c0525bc7d01f727)
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.12.10 | packaged by conda-forge | (main, Apr 10 2025, 22:21:13) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.11.0-26-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: AMD Radeon RX 9070 XT (gfx1201)
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: 6.4.43483
MIOpen runtime version: 3.4.0
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        48 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               12
On-line CPU(s) list:                  0-11
Vendor ID:                            AuthenticAMD
Model name:                           AMD Ryzen 5 7600X 6-Core Processor
CPU family:                           25
Model:                                97
Thread(s) per core:                   2
Core(s) per socket:                   6
Socket(s):                            1
Stepping:                             2
Frequency boost:                      enabled
CPU(s) scaling MHz:                   80%
CPU max MHz:                          5453.0000
CPU min MHz:                          545.0000
BogoMIPS:                             9381.75
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid overflow_recov succor smca fsrm flush_l1d amd_lbr_pmc_freeze
Virtualization:                       AMD-V
L1d cache:                            192 KiB (6 instances)
L1i cache:                            192 KiB (6 instances)
L2 cache:                             6 MiB (6 instances)
L3 cache:                             32 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-11
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; Safe RET
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] mypy==1.13.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.2
[pip3] onnx==1.17.0
[pip3] onnxscript==0.1.0.dev20240817
[pip3] optree==0.13.0
[pip3] torch==2.6.0+git684f6f2
[pip3] torchvision==0.21.0+4040d51
[pip3] triton==3.2.0
[conda] No relevant packages

**Comment by Yxeyuo** [comment]
looks like the formatting is broken I am sorry hopefully it is still readable if not tell me and I will try it again.
